### PR TITLE
adds dockerfile for prokka 1.14.6

### PR DESCRIPTION
**Pull request by kapsakcj**
Attempting to resolve issue #444 with blast binary incompatibilities with prokka.

@SESchmedes I've pushed this docker image to dockerhub, if you'd like to test out on your cluster. Hoping this resolves the blast issues 🙏 

(test) docker image name: `kapsakcj/prokka:1.14.6`
https://hub.docker.com/r/kapsakcj/prokka/tags

Setting this as a draft PR until we're confident the blast issues are ironed out.

This PR adds:

- a dockerfile for v1.14.6 (small patch release of prokka)
- prokka-specific README.md
- main README.md updated with new version

Dockerfile changes from previous version (1.14.5). This dockerfile was made quite a while ago, so it required some modernizing.

- base upgraded to `ubuntu:bionic` (xenial is EOL/in extended security maintainance mode)
- `tbl2asn` dependency installed via `apt`: `libidn11`
- added `PROGRAM_VAR` variables for blast, bedtools, barrnap (everything installed manually)
- install ncbi-blast+ manually instead of using binaries provided with `prokka` code
  - prioritize this install in `$PATH` over the prokka-provided-binaries
- added a test layer that runs the test commands from: https://github.com/tseemann/prokka/blob/master/.travis.yml

Pull Request (PR) checklist:
- [x] Include a description of what is in this pull request in this message.
- [x] The dockerfile successfully builds to a test target for the user creating the PR. (i.e. `docker build --tag samtools:1.15test --target test docker-builds/samtools/1.15` )
- [x] Directory structure as name of the tool in lower case with special characters removed with a subdirectory of the version number (i.e. `spades/3.12.0/Dockerfile`)
   - [x] (optional) All test files are located in same directory as the Dockerfile (i.e. `shigatyper/2.0.1/test.sh`)
- [x] Create a simple container-specific [README.md](https://github.com/StaPH-B/docker-builds/blob/master/.github/workflow-templates/readme-template.md) in the same directory as the Dockerfile (i.e. `spades/3.12.0/README.md`)
   - [x] If this README is longer than 30 lines, there is an explanation as to why more detail was needed
- [x] Dockerfile includes the recommended [LABELS](https://github.com/StaPH-B/docker-builds/blob/master/dockerfile-template/Dockerfile#L8-L18) 
- [x] Main [README.md](https://github.com/StaPH-B/docker-builds/blob/master/README.md) has been updated to include the tool and/or version of the dockerfile(s) in this PR
- [x] [Program_Licenses.md](https://github.com/StaPH-B/docker-builds/blob/master/Program_Licenses.md) contains the tool(s) used in this PR and has been updated for any missing

### CODE REVIEW

**Comment by kapsakcj** [comment]
@SESchmedes confirmed this updated docker image works without issues now 👍 Thanks again for testing!

So I've marked this PR ready for review

File: prokka/1.14.6/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+# run tests from https://github.com/tseemann/prokka/blob/master/.travis.yml
+RUN realpath . && prokka --version && \
+ prokka --help && \
+ ! prokka --doesnotexist && \
+ prokka --depends && \
+ prokka --setupdb && \
+ prokka --listdb && \
+ prokka --cpus 2 --outdir asm --prefix asm test/plasmid.fna && \
+ grep '>' asm/asm.fna && \
+ prokka --cleandb
```
Comment (erinyoung) [inline prokka/1.14.6/Dockerfile:127]: This test
- checks a lot of prokka options
- runs prokka on a test fasta

File: prokka/1.14.6/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+
+WORKDIR /data
+
+FROM app as test
+
+# re-instantiating for the test build layer
+ARG PROKKA_VER
+ARG BEDTOOLS_VER
+ARG BARRNAP_VER
+ARG BLAST_VER
```
Comment (erinyoung) [inline prokka/1.14.6/Dockerfile:114]: Are BEDTOOLS_VER, BARRNAP_VER, and BLAST_VER used in the test layer? I can't find where they're used.

**Comment by erinyoung** [comment]
I think this looks alright. I think there are some unused variables in the test layer, though. Could you remove those for readability?

File: prokka/1.14.6/Dockerfile (hunk 2 above)
Comment (kapsakcj) [reply prokka/1.14.6/Dockerfile:114]: no, I will remove them

**Review by erinyoung — APPROVED** [review]
Perfect!

Does this version overwrite the 'latest' tag?

**Comment by kapsakcj** [comment]
Yeah, I say we go ahead and overwrite the `:latest` tag since the minor release 1.14.5 -> 1.14.6 was a bug fix related to blast

**Comment by erinyoung** [comment]
I have merged this PR

I am going to deploy this to dockerhub and quay and overwrite the 'latest' tag.

Thank you for your contribution!

**Pull request merged**